platform/posix/utest/reconnect_utest.c

```diff
@@ -108,21 +120,49 @@ void test_Transport_ReconnectBackoffAndSleep_Succeeds( void )
 }
 
 /**
- * @brief Test that the value of the next max jitter is capped at some max threshold.
+ * @brief Test that the value of the next max jitter has a lower bound that will
+ * then be capped at some max threshold.
  */
-void test_Transport_ReconnectBackoffAndSleep_Jitter_Has_Max( void )
+void test_Transport_ReconnectBackoffAndSleep_Lower_Bound_Jitter_To_Cap( void )
 {
     bool retriesArePending;
 
     /* Initialize to 0 attempts, so the max value of next jitter will increase. */
-    reconnectParams.attemptsDone = 0;
+    reconnectParams.attemptsDone = 0U;
     reconnectParams.nextJitterMax = MAX_RECONNECT_BACKOFF_SECONDS / 2U;
 
     /* The next max jitter doubles on every retry unless it is greater than or
      * equal to half of the max threshold. */
+    rand_ExpectAndReturn( 0 );
+    sleep_ExpectAnyArgsAndReturn( 0 );
+    retriesArePending = Transport_ReconnectBackoffAndSleep( &reconnectParams );
+    TEST_ASSERT_TRUE( retriesArePending );
+    TEST_ASSERT_EQUAL( reconnectParams.nextJitterMax,
+                       MAX_RECONNECT_BACKOFF_SECONDS );
+    TEST_ASSERT_LESS_THAN_UINT32( MAX_RECONNECT_ATTEMPTS,
+                                  reconnectParams.attemptsDone );
```

Comment:
This should test for the exact expected value of `attemptsDone`
```suggestion
    TEST_ASSERT_EQUAL( 1u, reconnectParams.attemptsDone );
```